### ISSUE TITLE
Mpeg - mp4 Works but only until 2000 frames is ok now

### ISSUE THREAD

**Issue by Laniaktroid** [jackburton79/bescreencapture#50]
Hello there, i was trying the new features of bescreencapture, and it work better now but still have a similar limit than the last time i did report wth more than 2000-2200-+ frames, if it record more frames give an encoding error. around more than 6 minutes(?

**Comment by jackburton79** [comment]
Hi,
will try with more than 2000 frames.
is it full screen ?

**Comment by Laniaktroid** [comment]
Yes, it is a full screen record, i did'nt see that answer before sorry, but yes, i am trying to do some gameplays on haiku, but it cant record more than 2000 frames, similar to how it was before but a little more frames.

**Comment by jackburton79** [comment]
I managed to record more than 2000 frames once or twice, but I definitely had some issues at least once.
I know that it takes some time to record so many frames, but what happens if you select "GIF" or "BMP" as file format ?
This seems to work every time, so I'm starting to think there could be an issue in the encoding part

**Comment by Laniaktroid** [comment]
![image](https://user-images.githubusercontent.com/12539219/164957283-f64ec923-87df-44a9-8ede-c4c13243c21c.png)
![image](https://user-images.githubusercontent.com/12539219/164957284-5bbe3cf7-eb65-4e4d-b336-74a77d50d54b.png)


Those screenshots are for a video with more than 3000 frames, at the end it could not create the clip, this is the last version on the repository, later i will try with gif

**Comment by Laniaktroid** [comment]
![image](https://user-images.githubusercontent.com/12539219/165007802-1d02ac05-4c07-482b-bbc3-01b6ba69c69a.png) the GIF format the "exporting" process never end, it take more than an hour with no end at this point.

**Comment by jackburton79** [comment]
I've debugged this a bit.
It fails in MovieEncoder:_WriteFrame(), on the line fMediaFile->CommitHeader().
For some reason, when you record more than 3000 frames, this fails with B_NO_INIT.
This smells like memory corruption

**Comment by jackburton79** [comment]
It was caused by setting the field rate to a value the media kit does not like. I disabled setting it for now.

Should be fixed in master.
Could you try, please ?

**Comment by jackburton79** [comment]
Fixed (in a better way) in master, with 9aa52e288e89e63ad0aea580cc00551e14678c06
